kart/schema.py

```diff
@@ -237,18 +239,25 @@ def interpretation(self):
         return self.get("interpretation")
 
     @property
-    def id_or_name_or_interpretation(self):
+    def best_identifier(self):
         """
         Different types of schemas have these various ways of identifying themselves.
         When we do schema diffs, it makes them more readable if we match up the columns using this property.
+        At a last resort, we use the hash of the entire column contents as an "ID".
         """
         if "id" in self:
             return self.get("id")
         if "name" in self:
             return self.get("name")
         if "interpretation" in self:
             return self.get("interpretation")
```

Comment:
have you come across any TIF files with missing color interpretations? that's an optional thing in images. I think GDAL will give you the string "Undefined" for missing color interpretations; might be worth testing it.

IMO this isn't unusual and we probably need to handle it or at least have a plan for handling it  - basically don't bank on interpretation being set.

Technically interpretation may be present but be non-unique too - eg you can have multiple red bands) - in that case the only valid identifier might be the band's index (band 0, band 1 etc). This is probably pretty unusual though, so we probably don't need to handle it at present

Reply:
I haven't looked very hard for weird TIF files yet. But yes I'm assuming that interpretation - as extracted by Kart - could be missing, and there could be more than one column which has a missing interpretation.
I haven't allowed for there being non-null non-unique interpretations - eg two reds, or perhaps more sensibly, two with an interpretation of "categoryIndex" which index into two different category tables.
I'll just add some notes to the tracking bug